text/0000-lifecycle-multicall-binary-for-build.md

```diff
@@ -0,0 +1,78 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle as a multicall binary for build phases
+- Start Date: 01/21/2020
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+We should compile the lifecycle as a multicall binary for the build phases of detect, analyze, restore, build, export and rebase while retaining the interface to the phases. The binary can also support the phases as subcommands with the phases in their verb form.
+
+
+# Motivation
+[motivation]: #motivation
+
+- The size of the lifecycle release decreases from 24M to 7M.
+- This should speed up time take to build the lifecycle as far fewer binaries get built.
+- The size of the builders will decrease.
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+We are proposing changing the lifecycle release archive to contain the following:
+- `lifecycle` binary: The build phases (detector, analyzer, restorer, builder, exporter, rebaser) of the lifecycle will exist as a multicall binary
+- symlinks linking `detector`, `analyzer`, `restorer`, `builder`, `exporter` and `rebaser` to the `lifecycle` binary
+- `launcher` binary: The launcher will exist as a separate binary
```

Comment:
Is the reason for this because it isn't part of the "build" process?

Reply:
This is because the launcher get's added to the exported image. It doesn't seem right to add the entire lifecycle to the exported image.